hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -191,7 +191,8 @@ public DistributedObjectFuture getOrCreateProxyFuture(String name, boolean publi
      * @param initialize   true if he DistributedObject proxy object should be initialized.
      * @return The DistributedObject instance if it is created by this method, null otherwise.
      */
-    public DistributedObjectFuture createProxy(String name, boolean publishEvent, boolean initialize) {
+    public DistributedObjectFuture createProxy(String name, boolean publishEvent, boolean initialize,
+                                               boolean local) {
```

Comment:
Aren't `publishEvent ` and `local ` always the opposite of each other? `assert publishEvent == !local`

Reply:
I think we can remove `publishEvent` argument completely and use `!local` to decide whether or not to publish the event.

Reply:
yes, I agree